### PR TITLE
chore(windows): move common components to common 🎢

### PR DESCRIPTION
**Pull request by mcdurdin**
@keymanapp-test-bot skip

This PR's goal is to move the following files in windows/src/global/delphi/comp to common:

* ExtShiftState.pas (to common/windows/delphi/general)
* FixedTrackbar.pas
* keybtn.pas
* OnScreenKeyboard.pas
* PaintPanel.pas

All other files moved are dependencies of these files, into common/windows/delphi subfolders; remaining files in windows/src/global/delphi/general, windows/src/global/delphi/vcl will be moved in upcoming PRs.

Also:
* Added a common_components package for the moved components and build infrastructure for it.
* Updated all project references to moved files.
* Fixed up the build for keymanversion_build.in (global_versions target in Makefiles)

### CODE REVIEW

File: common/windows/delphi/ext/sentry/test/SentryClientVclTest.dpr [hunk 1]
```diff
@@ -7,13 +7,13 @@ uses
   sentry in '..\sentry.pas',
   Sentry.Client.Vcl in '..\Sentry.Client.Vcl.pas',
   Keyman.System.KeymanSentryClient in '..\..\..\global\delphi\general\Keyman.System.KeymanSentryClient.pas',
-  KeymanVersion in '..\..\..\global\delphi\general\KeymanVersion.pas',
+  KeymanVersion in '..\..\..\..\..\common\windows\delphi\general\KeymanVersion.pas',
```
Comment (ermshiperete) [inline common/windows/delphi/ext/sentry/test/SentryClientVclTest.dpr:10]: I don't think `common/common/windows` is correct. Strange that it still compiles...

I think the correct path is `..\..\..\..\..\..\common\windows\delphi\general\KeymanVersion.pas` or `..\..\..\..\..\windows\delphi\general\KeymanVersion.pas` which can be shortened to `..\..\..\general\KeymanVersion.pas`

```suggestion
  KeymanVersion in '..\..\..\general\KeymanVersion.pas',
```

File: common/windows/delphi/ext/sentry/test/SentryClientVclTest.dpr [hunk 2]
```diff
@@ -7,13 +7,13 @@ uses
...
   KeymanPaths in '..\..\..\global\delphi\general\KeymanPaths.pas',
-  RegistryKeys in '..\..\..\global\delphi\general\RegistryKeys.pas',
+  RegistryKeys in '..\..\..\..\..\common\windows\delphi\general\RegistryKeys.pas',
```
Comment (ermshiperete) [inline common/windows/delphi/ext/sentry/test/SentryClientVclTest.dpr:12]:
```suggestion
  RegistryKeys in '..\..\..\general\RegistryKeys.pas',
```

File: common/windows/delphi/ext/sentry/test/SentryClientVclTest.dpr [hunk 3]
```diff
@@ -7,13 +7,13 @@ uses
...
   DebugPaths in '..\..\..\global\delphi\general\DebugPaths.pas',
-  ErrorControlledRegistry in '..\..\..\global\delphi\vcl\ErrorControlledRegistry.pas',
+  ErrorControlledRegistry in '..\..\..\..\..\common\windows\delphi\vcl\ErrorControlledRegistry.pas',
```
Comment (ermshiperete) [inline common/windows/delphi/ext/sentry/test/SentryClientVclTest.dpr:14]:
```suggestion
  ErrorControlledRegistry in '..\..\..\vcl\ErrorControlledRegistry.pas',
```

File: common/windows/delphi/ext/sentry/test/SentryClientVclTest.dpr [hunk 4]
```diff
@@ -7,13 +7,13 @@ uses
...
   utilexecute in '..\..\..\global\delphi\general\utilexecute.pas',
-  Unicode in '..\..\..\global\delphi\general\Unicode.pas';
+  Unicode in '..\..\..\..\..\common\windows\delphi\general\Unicode.pas';
```
Comment (ermshiperete) [inline common/windows/delphi/ext/sentry/test/SentryClientVclTest.dpr:16]:
```suggestion
  Unicode in '..\..\..\general\Unicode.pas';
```

File: common/windows/delphi/ext/sentry/test/SentryClientVclTest.dproj [hunk 5]
```diff
@@ -116,13 +116,13 @@
         <DCCReference Include="..\sentry.pas"/>
         <DCCReference Include="..\Sentry.Client.Vcl.pas"/>
         <DCCReference Include="..\..\..\global\delphi\general\Keyman.System.KeymanSentryClient.pas"/>
-        <DCCReference Include="..\..\..\global\delphi\general\KeymanVersion.pas"/>
+        <DCCReference Include="..\..\..\..\..\common\windows\delphi\general\KeymanVersion.pas"/>
```
Comment (ermshiperete) [inline common/windows/delphi/ext/sentry/test/SentryClientVclTest.dproj:119]:
```suggestion
        <DCCReference Include="..\..\..\general\KeymanVersion.pas"/>
```

File: common/windows/delphi/ext/sentry/test/SentryClientVclTest.dproj [hunk 6]
```diff
@@ -116,13 +116,13 @@
...
         <DCCReference Include="..\..\..\global\delphi\general\KeymanPaths.pas"/>
-        <DCCReference Include="..\..\..\global\delphi\general\RegistryKeys.pas"/>
+        <DCCReference Include="..\..\..\..\..\common\windows\delphi\general\RegistryKeys.pas"/>
```
Comment (ermshiperete) [inline common/windows/delphi/ext/sentry/test/SentryClientVclTest.dproj:121]:
```suggestion
        <DCCReference Include="..\..\..\general\RegistryKeys.pas"/>
```

File: windows/src/test/manual-tests/test_iconsave/test_iconsave.dpr [hunk 7]
```diff
@@ -10,10 +10,10 @@ uses
   crypt_guid in '..\..\global\delphi\productactivation\crypt_guid.pas',
   productactivation in '..\..\global\delphi\productactivation\productactivation.pas',
   tavultesoft_certificate in '..\..\global\delphi\productactivation\tavultesoft_certificate.pas',
-  Unicode in '..\..\global\delphi\general\Unicode.pas',
+  Unicode in '..\..\..\..\common\windows\delphi\general\Unicode.pas',
```
Comment (ermshiperete) [inline windows/src/test/manual-tests/test_iconsave/test_iconsave.dpr:13]:
```suggestion
  Unicode in '..\..\..\..\..\common\windows\delphi\general\Unicode.pas',
```

File: windows/src/test/manual-tests/test_iconsave/test_iconsave.dpr [hunk 8]
```diff
@@ -10,10 +10,10 @@ uses
...
   crypt_user in '..\..\global\delphi\productactivation\crypt_user.pas',
   KeyNames in '..\..\global\delphi\general\KeyNames.pas',
-  KeymanVersion in '..\..\global\delphi\general\KeymanVersion.pas',
+  KeymanVersion in '..\..\..\..\common\windows\delphi\general\KeymanVersion.pas',
```
Comment (ermshiperete) [inline windows/src/test/manual-tests/test_iconsave/test_iconsave.dpr:16]:
```suggestion
  KeymanVersion in '..\..\..\..\..\common\windows\delphi\general\KeymanVersion.pas',
```

File: windows/src/test/manual-tests/test_iconsave/test_iconsave.dproj [hunk 9]
```diff
@@ -96,10 +96,10 @@
 			<DCCReference Include="..\..\global\delphi\productactivation\crypt_guid.pas"/>
 			<DCCReference Include="..\..\global\delphi\productactivation\productactivation.pas"/>
 			<DCCReference Include="..\..\global\delphi\productactivation\tavultesoft_certificate.pas"/>
-			<DCCReference Include="..\..\global\delphi\general\Unicode.pas"/>
+			<DCCReference Include="..\..\..\..\common\windows\delphi\general\Unicode.pas"/>
```
Comment (ermshiperete) [inline windows/src/test/manual-tests/test_iconsave/test_iconsave.dproj:99]:
```suggestion
			<DCCReference Include="..\..\..\..\..\common\windows\delphi\general\Unicode.pas"/>
```

**Review by ermshiperete — CHANGES_REQUESTED** [review]

**Comment by mcdurdin** [comment]
Many of the manual tests will fail to build as they are very, very old. We could go and clean them up but it is a low-value task. We could even go and delete nearly all of them. But I don't want to do that in this PR chain.

I've fixed a couple which were more recent. The others, I propose to leave and not fix until we actually need them.

**Pull request merged**